### PR TITLE
Added new properties to localStorage user, fixed region picker, fixed…

### PR DESCRIPTION
**Pull request by kevinylee1991**
… regions and tokens in simulator

Still need to change old UI to use the newer 'user-preferences-<username>' localStorage property with last used tenant/region data.

### CODE REVIEW

File: src/api-client/Keystone.js [hunk 1]
```diff
@@ -103,12 +103,36 @@ class Keystone {
     try {
       const response = await axios.post(this.tokensUrl, body)
       const scopedToken = response.headers['x-subject-token']
+      const _user = response.data.token.user
+      const roles = response.data.token.roles
+
+      const matchRole = (roles, roleName) => {
```
Comment (brennancheung) [inline src/api-client/Keystone.js]: `const matchRole = (roles, roleName) => roles.some(propEq('name'))`

File: src/api-client/Keystone.js [hunk 2]
```diff
@@ -103,12 +103,36 @@ class Keystone {
...
+        return roles.some((role) => role && role.name === roleName)
+      }
+
+      const role = roles.reduce((acc, current) => {
```
Comment (brennancheung) [inline src/api-client/Keystone.js]: This makes it very hard to understand the logic.  I'm not sure what you are doing here.  But maybe a `map` followed by a `uniq` might make more sense here?

File: src/api-client/Keystone.js (hunk 1 above)
Comment (brennancheung) [reply src/api-client/Keystone.js]: Also maybe `const containsRoles = (roles, roleName) => pluck('name', roles).contains(roleName)`

Comment (brennancheung) [reply src/api-client/Keystone.js]: contains is a little clearer than match

File: src/api-client/Keystone.js (hunk 2 above)
Comment (kevinylee1991) [reply src/api-client/Keystone.js]: I added a comment above the code block, but I wasn't able to think of a clearer solution that doesn't add steps

File: src/api-client/Keystone.js [hunk 3]
```diff
@@ -103,12 +104,25 @@ class Keystone {
     try {
       const response = await axios.post(this.tokensUrl, body)
       const scopedToken = response.headers['x-subject-token']
+      const _user = response.data.token.user
+      const roles = response.data.token.roles
+
+      const roleNames = roles.map((role) => role.name)
```
Comment (brennancheung) [inline src/api-client/Keystone.js]: `const roleNames = pluck('name', roles)`

**Pull request merged**